docs/development/testing.md

```diff
@@ -1,204 +1,54 @@
-
-```bash
-KOPS_VERSION=`bazel run //cmd/kops version -- --short`
-export DOCKER_IMAGE_PREFIX=${USER}/
-export DOCKER_REGISTRY=
-make kops-controller-push
-export KOPSCONTROLLER_IMAGE=${DOCKER_IMAGE_PREFIX}kops-controller:${KOPS_VERSION}
+go test -v ./...
 ```
 
-You can create a cluster using `kops create cluster <clustername> --zones us-east-1b`
+### Adding an integration test
 
-Then follow the test directions above.
+The integration tests takes a cluster spec and builds cloudformation/terraform templates. To add a new integration test, create a new directory in `tests/integration/update_cluster/` and put the cluster spec in `in-v1alpha2.yaml`. Use a unique cluster name.
 
-To override the test list for the job, you need to familiar with the
-`ginkgo.focus` and `ginkgo.skip`
-flags. Using these flags, you can do:
+Then edit `./cmd/kops/integration_test.go` and add the test function with the cluster name and directory from above.
 
-```
-export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Performance\]"
-```
+Lastly run `hack/update-expected` to generate the expected output.
 
-and follow the instructions above. [Here are some other examples from the `e2e.go` documentation.](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md).
+## Kubernetes e2e testing
 
-If you want to test against an existing cluster, you can do:
+### Preparing the environment
 
-```
-export E2E_UP=false; export E2E_DOWN=false
-```
+The easiest way to run the Kubernetes e2e tests is to install the `kubetest` binary from test-infra.
 
-and follow the instructions above. This is particularly useful for testing the
-myriad of `kops` configuration/topology options without having to modify the
-testing infrastructure. *Note:* This is also the only way currently to test a
-custom Kubernetes build
-(see
-[kubernetes/test-infra#1454](https://github.com/kubernetes/test-infra/issues/1454)).
+See [e2e-tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md) for instructions how to install and general usage of the utility.
 
+You can see https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/kops/ the various jobs that are periodically run against kops.
+The container arguments for each of the jobs are the ones passed to `kubetest`.
 
-### Uploading a custom build
+Following the examples below, kubetest will download artifacts such as a given Kubernetes build. Therefore you probably want to run `kubetest` from a directory dedicated for this purpose.
 
-If you want to upload a custom Kubernetes build, here is a simple way (note:
-this assumes you've run `make quick-release` in the Kubernetes repo first):
+### Running against an existing cluster
 
+If you want `kubetest` to run spin up and tear down the cluster on every run, you can run something like the following to re-use an existing cluster.
+This assumes you have already built the kops binary from source.
```

Comment:
i think kops' `make bazel-build-cli` puts the binary in `./bazel-bin/cmd/kops/$GOOS_$GOARCH_pure_stripped/kops` so maybe its worth at least mentioning something like that.

```suggestion
This assumes you have already built the kops binary from source. The exact path to the `kops` binary used in the `--kops` flag may differ.
```